grafana/grafana
# logs: fix scanning bug

if a logs-search returns no logs we display the "scan for older logs" button.
when you press it, while the scanning happens, we display the "stop scan" button.

there is currently a bug that causes the "stop scan" button vanish immediately, the PR fixes that bug. the bug happens because as soon as the scan-request starts, you get a "response" which is basically `loadingstate.loading`, and that causes the code to call `scanStopAction` immediately

how to test:
    - you need to be able to make a slow logs-database-response somehow
    - make a query that returns no results
    - press the "scan for older logs" button
    - the scanning should start, and you should be able to see the "stop scan" button
    - press the "stop scan" button
    - the "stop scan" button should vanish


(NOTE: originally i had one more functionality in this PR (add tracking to the buttons), but i moved that change to a separate PR: https://github.com/grafana/grafana/pull/71031)
